frontend/src/lib/ParserUtils.ts

```diff
@@ -14,3 +14,20 @@ export function parseTaskDisplayName(metadata?: Metadata): string | undefined {
   }
   return taskDisplayName || componentDisplayName;
 }
+
+export function parseNodeDisplayName(nodeId: string, workflow?: Workflow): string {
+  const node = workflow?.status.nodes[nodeId];
+  if (!node) {
+    return nodeId;
+  }
+  const workflowName = (workflow?.metadata && workflow?.metadata.name) || '';
```

Comment:
nit: this can be simplified as
```suggestion
  const workflowName = workflow?.metadata?.name || '';
```